R/baselinenowcast.R

```diff
@@ -0,0 +1,151 @@
+#'    default estimate uncertainty using past retrospective nowcast errors and
+#'    generate probabilistic nowcasts, which are samples from the predictive
+#'    distribution of the estimated final case count at each reference date.
+#'    This method specifically computes a nowcast for a single reporting
+#'    triangle. See documentation for the arguments of this function which
+#'    can be used to set the model specifications (things like number of
+#'    reference times for delay and uncertainty estimation, the observation
+#'    model, etc.).
+#'
+#' @param data \code{\link{reporting_triangle}} class object to be nowcasted.
+#' @param delay_pmf Vector of delays assumed to be indexed starting at the
+#'   first delay column in `data$reporting_triangle_matrix`. Default is NULL,
+#'   which will estimate the delay from the reporting triangle matrix in `data`,
+#'   See \code{\link{estimate_delay}} for more details.
+#' @param uncertainty_params Vector of uncertainty parameters ordered from
+#'   horizon 1 to the maximum horizon. Default is `NULL`, which will
+#'   result in computing the uncertainty parameters from the reporting
+#'   triangle matrix `data`. See \code{\link{estimate_uncertainty}} for more
+#'   details.
+#' @param ... Additional arguments passed to
+#'    \code{\link{estimate_uncertainty}}
+#'    and \code{\link{sample_nowcast}}.
+#' @inheritParams baselinenowcast
+#' @inheritParams estimate_uncertainty
+#' @inheritParams sample_nowcast
+#' @inheritParams allocate_reference_times
+#' @family baselinenowcast_df
+#' @export
+#' @method baselinenowcast reporting_triangle
+#' @returns Data.frame of class \code{\link{baselinenowcast_df}}
+#' @examples
+#' data_as_of_df <- syn_nssp_df[syn_nssp_df$report_date <= "2026-04-01", ]
+#' rep_tri <- as_reporting_triangle(
+#'   data = data_as_of_df,
+#'   max_delay = 25
+#' )
+#' nowcast_df <- baselinenowcast(rep_tri)
+#' nowcast_df
+baselinenowcast.reporting_triangle <- function(
+    data,
+    scale_factor = 3,
+    prop_delay = 0.5,
+    output_type = "samples",
+    draws = 1000,
+    uncertainty_model = fit_by_horizon,
+    uncertainty_sampler = sample_nb,
+    delay_pmf = NULL,
+    uncertainty_params = NULL,
+    ...) {
+  tri <- data$reporting_triangle_matrix
+  assert_choice(output_type, choices = c("samples", "point"))
+  assert_integerish(draws, null.ok = TRUE)
+
+  tv <- allocate_reference_times(tri,
+    scale_factor = scale_factor,
+    prop_delay = prop_delay
+  )
+
+  if (is.null(delay_pmf)) {
+    delay_pmf <- estimate_delay(
+      reporting_triangle = data$reporting_triangle_matrix,
+      n = tv$n_history_delay
+    )
+  }
+  # check for delay pmf being the right length/format
+  .validate_delay(tri, delay_pmf)
+
+  pt_nowcast <- apply_delay(tri, delay_pmf)
+
+  if (output_type == "point") {
+    nowcast_df <- data.frame(
+      time = seq_len(nrow(pt_nowcast)),
+      pred_count = rowSums(pt_nowcast)
+    )
+    if (!is.null(uncertainty_params)) {
```

Comment:
I really don't think you need this edge case check. I would remove and add if you get user reports of an issue.